pulsar-client-kafka-compat/pulsar-client-kafka/src/test/java/org/apache/pulsar/client/kafka/compat/tests/KafkaConsumerTest.java

```diff
@@ -69,6 +69,7 @@ public void testSimpleConsumer() throws Exception {
 
         Consumer<String, String> consumer = new PulsarKafkaConsumer<>(props);
         consumer.subscribe(Arrays.asList(topic));
+        Thread.sleep(pulsar.getConfiguration().getReadOnActiveConsumerDelayTimeMillis() + 500);
```

Comment:
Is there any reason for adding `Thread.sleep()` here? if not then can we avoid it?

Reply:
If I understand correctly, the subscription type of PulsarKafkaConsumer is Failover.
https://github.com/apache/incubator-pulsar/blob/e7f1160f563e433f1bb2ec70da827c7d420c23d2/pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/consumer/PulsarKafkaConsumer.java#L191

As the result of this modification, a Failover consumer can not receive any messages for at least 1000ms after starting subscription.
This test fails more often than not because the receive timeout is set to 100ms.
https://github.com/apache/incubator-pulsar/blob/e7f1160f563e433f1bb2ec70da827c7d420c23d2/pulsar-client-kafka-compat/pulsar-client-kafka/src/test/java/org/apache/pulsar/client/kafka/compat/tests/KafkaConsumerTest.java#L82

Reply:
I think there are two options:

1. Set `readOnActiveConsumerDelayTimeMillis` on the broker side to 0.
2. Increase the receive timeout on the consumer side to 1000ms or more.

Reply:
yes, we should go with 1st option where by default we can keep `readOnActiveConsumerDelayTimeMillis=0` so, it won't affect any test. we can set it in [MockedPulsarServiceBaseTest](https://github.com/apache/incubator-pulsar/blob/master/pulsar-broker/src/test/java/org/apache/pulsar/broker/auth/MockedPulsarServiceBaseTest.java#L86) also